### ISSUE TITLE
GoogleCloudMessageについて

### ISSUE THREAD

**Issue by KOROKORO-Manbo** [NIFTYCloud-mbaas/UserCommunity#538]
Androidのプッシュ通知の解説ページではまだGoogleCloudMessagingでの説明になっていますが今のGoogleのAPI設定ではFirebaseのものしか使えないようになっている漢字なのですが、前からあったGCMのAPIって今でも有効にでき、認証キーを得ることはできるのでしょうか？

**Comment by kogorou** [comment]
同じ問題で困っております
少なくともプッシュ通知での設定では、
NCMBInstallation#getRegistrationIdInBackground の第１引数に入れるべき値を
今後どのようにすればよいか わからず悩んでおります。。。

**Comment by kojiro-goto-vc** [comment]
https://console.developers.google.com
API Managerに行って、左のライブラリを選択し、検索窓に"GCM"を入れると、GCMの管理画面のリンクが出るのでそこから、有効化して、あとはサーバーキーを作れば良いだけですね。
そうすれば、今まで通りのsender id、api keyが使えます。
getRegistrationIdInBackgroundに入れるのはsender idで、
http://mb.cloud.nifty.com/doc/current/push/basic_usage_android.html#配信端末情報の登録
にあるように、

> YOUR_PROJECT_NUMBERの部分には、Google Developer Consoleで作成したプロジェクトのProject Numberを設定します。

でできると思います。

**Comment by KOROKORO-Manbo** [comment]
ありがとうございます。設定ページを見つけられました。ただGoogleからのメールを読む限り、いつまでGCMの方をサポートしてくれるかすごい不安ですが

**Comment by kojiro-goto-vc** [comment]
そうですね。mBaasとかParseベースで作っていた人、その他有料のBaasを提供している人たちは結構厳しい状況になるのかな？と思ったりもしますが、試しにFirebaseに適当なアプリを作ってみたら、アプリの設定画面に「クラウドメッセージング」っていうタブがあり、そこに今までのGCMと同じ、サーバーキーと送信者ID（GCMの時のプロジェクト番号ですかね）がありました。これをNCMBに設定すれば、GCMが使えるなじゃないかな？と思っていますが、試してません。
FCMのサーバーの説明部分はGCM時代とほとんど変わっていないと思うので、あとはFCMのSDKとどうやって、NCMBのSDKを連携させるかということになるかと思います。
NCMB内でRegistration IDを取得しているので、その部分を変更しないとダメかもしれませんね。
Niftyの公式見解が欲しいところです。
